packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -483,6 +496,10 @@ class DateTimePicker extends React.Component {
       // again (unless they change the DateTime).
       this.props.onChange(event, newDateTime && newDateTime.isValid() ? newDateTime.format() : '');
     }
+
+    // Handle blur with the new selection upon dismiss of the Time Clarification dialog.
+    this.handleBlur(event, newDateTime);
```

Comment:
Did some testing using this deployed [example](https://terra-framework-deploye-pr-784.herokuapp.com/#/components/terra-date-time-picker/date-time-picker/events). onBlur is called once after closing the clarification dialog for the first time This appears correct. However, if I click on the offset button to open the clarification dialog and then close it, both onFocus and onBlur get called. When clicking anywhere else on the page to lose focus from the entire component, onBlur is called once again, which ends up getting called twice.

Here are the steps:
1. Go to [example](https://terra-framework-deploye-pr-784.herokuapp.com/#/components/terra-date-time-picker/date-time-picker/events) and keep track of the "onBlur Trigger Count"
2. Select 11/03/2019 as the date and 01:30 as the time to create the ambiguous hour.
3. Lose focus.
4. Select either button in the clarification dialog to close the dialog.
5. onBlur is called and the onBlur trigger count becomes 1.
6. Click outside of the entire date-time component to lose focus.
7.  onBlur is not called and the onBlur trigger count remains 1.
8. Everything up to this point appears correct.
9. Click on the offset button to relaunch the clarification dialog.
10. Select either button in the clarification dialog to close the dialog.
11. onBlur is called and the onBlur trigger count becomes 2.
12. Click outside of the entire date-time component to lose focus.
13. onBlur is called again and the onBlur trigger count remains 3. This is inconsistent with step 7 and ends up triggering onBlur twice when the consumer expects it to get triggered just once.

Reply:
Ah, I see what is happening - Step 3 loses focus and that gets correctly handled once the Time Clarification dialog is dismissed (ambiguity is resolved in Step 4) and at this point there is no focus back in the date time picker component. 

However, when the clarification dialog is shown by explicitly clicking on the offset button, the focus is not really lost and blur should not be handled because the focus goes back to the offset button when the dialog is dismissed. 

I added a [focus outline](https://github.com/cerner/terra-framework/blob/master/packages/terra-date-time-picker/src/_TimeClarification.module.scss#L61) for in-house testing and here's what I noticed - the button still has focus in Step 11. This makes me think that the blur should not handled when the Time Clarification dialog was opened by clicking on the offset button. 

What are your thoughts on doing something like below?

```
// Handle blur with the new selection upon dismiss of the Time Clarification dialog.
if (!this.wasOffsetButtonClicked) {
   this.handleBlur(event, newDateTime);
}
```

By doing this, the blur handler will get called (correctly) when the user clicks outside of the entire date-time component to lose focus in Step 12. 

![onBlur 003](https://user-images.githubusercontent.com/10090859/62279935-a1766480-b468-11e9-85cb-7b6f4a56aa30.png)

Reply:
@nramamurth Adding that check to both handleStandardTimeButtonClick and handleDaylightSavingButtonClick does fix the issue. I think it makes sense for adding this check since the wrapper date-time component hasn't lost focus as the focus is still on the offset button. And then onBlur would get called when the user manually clicks outside of the wrapper component. Could you also update the comment above the check for why this check is needed?

Reply:
Right, fixed it here: https://github.com/cerner/terra-framework/pull/784/commits/1eaae53d54c7058dd14309b33fd5e0879cea9dd4